src/wifi.rs

```diff
@@ -1158,8 +1173,12 @@ impl<'d> WifiDriver<'d> {
     fn get_ap_conf(&self) -> Result<AccessPointConfiguration, EspError> {
         let mut wifi_config: wifi_config_t = Default::default();
         esp!(unsafe { esp_wifi_get_config(wifi_interface_t_WIFI_IF_AP, &mut wifi_config) })?;
-
-        let result: AccessPointConfiguration = unsafe { Newtype(wifi_config.ap).into() };
+        let access_point_configuration: Option<AccessPointConfiguration>;
+        unsafe {
+            access_point_configuration = Newtype(wifi_config.ap).into();
+        }
+        //unwrap should be safe here, as it should be impossible to configure a accesspoint, that cannot be represented
+        let result: AccessPointConfiguration = unsafe { access_point_configuration.unwrap() };
```

Comment:
Why the `unsafe` here? There is nothing `unsafe` in unwrapping an `Option`. It might panic, but that's all.

Reply:
Anyway, here also this change would be unnecessary, once you restore the original `From` definition for `AccessPointConfiguration`.